File: android/app/src/main/java/com/alexivanov/snapsell/domain/Condition.kt
package com.alexivanov.snapsell.domain

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** Item condition. Wire names match contracts/item.schema.json exactly. */
@Serializable
enum class Condition(val wire: String, val label: String) {
    @SerialName("new") NEW("new", "New"),
    @SerialName("like_new") LIKE_NEW("like_new", "Like new"),
    @SerialName("good") GOOD("good", "Good"),
    @SerialName("fair") FAIR("fair", "Fair"),
    @SerialName("for_parts") FOR_PARTS("for_parts", "For parts");

    companion object {
        fun fromWire(value: String): Condition =
            entries.firstOrNull { it.wire == value }
                ?: throw IllegalArgumentException("Unknown condition: $value")
    }
}

File: ios/SnapSell/Domain/Condition.swift
import Foundation

/// Item condition. Wire names match contracts/item.schema.json exactly.
enum Condition: String, Codable, CaseIterable, Identifiable {
    case new
    case likeNew = "like_new"
    case good
    case fair
    case forParts = "for_parts"

    var id: String { rawValue }
    var wire: String { rawValue }

    var label: String {
        switch self {
        case .new: return "New"
        case .likeNew: return "Like new"
        case .good: return "Good"
        case .fair: return "Fair"
        case .forParts: return "For parts"
        }
    }

    static func fromWire(_ value: String) throws -> Condition {
        guard let c = Condition(rawValue: value) else { throw AppError("Unknown condition: \(value)") }
        return c
    }
}
